toolbox/fdc3-conformance/FDC3-1.2-Conformance-Test-Cases.md

```diff
@@ -0,0 +1,191 @@
+# FDC3 1.2 Conformance Test Cases
+
+## 1. Basic Tests
+
+_These are some basic sanity tests implemented in the FDC3 Conformance Framework.  It is expected that Desktop Agent testers will run these first before commencing the much more thorough tests in section 2 onwards._
+
+- `BasicCL1`: You can call the `fdc3.addContextListener` with the `fdc3.contact` context type. The returned listener object has an `unsubscribe` function.
+- `BasicCL2`: You can call the `fdc3addContextListener` with no context type. The returned listener object has an `unsubscribe` function.
+- `BasicIL1`: You can call the `fdc3.addIntentListener` on the `DesktopAgent` for an intent, and get back a `Listener` object with `unsubscribe` method.
+- `BasicCH1`: A call to `fdc3.getCurrentChannel` on the `DesktopAgent` always returns a promise.
+- `BasicCH2`: A call to `fdc3.getCurrentChannel()` returns a promise resolving to _null_ if called prior to any `joinChannel`.
+- `BasicGI1`: A call to `fdc3.getInfo()` returns an object with `fdc3Version` and `provider` properties.
+- `BasicAC1`: A call to `fdc3.getOrCreateChannel(<name>)` will return an promise resolving to an object matching the `Channel` interface, with properties of `id`, `type`, `broadcast`, `getCurrentContext` and `addContextListener`.
+- `BasicUC1`: You can call the `fdc3.getSystemChannels()` function and receive a promise containing an array of more than 1 `Channel` objects, each with `type` and `id` set.
+- `BasicJC1`: You can call `fdc3.joinChannel`, passing in the `id` of one of the system channels.  After the returned promise is resolved, `fdc3.getCurrentChannel` should then return that joined channel.
```

Comment:
@robmoffat Kalina's PR is adding a BasicJC2 and BasicJC3 - BasicJC2 makes sense and needs adding here, but we can't do BasicJC3 (for the same reason others don't want to test OpenError.AppTimeout, it isn't clear that it could/should be returned if the channel was not in system channels). In fact *most* implementations of 1.2 allow you to join AppChannels - it was only in 2.0 that we started to clarify the intention that it should only be user channels that you join (and documented the relevant error).